### ISSUE TITLE
Add alignment and padding to type hover info

### ISSUE THREAD

**Issue by medhefgo** [clangd/clangd#1763]
Currently, when hovering over a type definition, it only contains its size. It would be helpful if it contained the type's alignment and (cumulative) padding.

```
struct A {
        char a;
        short b;
};
```

Would then show a hover like this:
```
struct A
Size: 4 bytes
Alignment: 2 bytes
Padding: 1 byte
```

**Comment by sr-tream** [comment]
Currently hover info contains offset, size and padding
![image](https://github.com/clangd/clangd/assets/12231048/b3d075de-0b79-41d1-8da9-60b61fc94bee)

**Comment by medhefgo** [comment]
> Currently hover info contains offset, size and padding

Yes, for the fields. Not the struct itself.

**Comment by sr-tream** [comment]
https://github.com/llvm/llvm-project/pull/67213

**Comment by medhefgo** [comment]
> llvm/llvm-project#67213

That still doesn't show the total padding of the struct? :(

**Comment by sr-tream** [comment]
> > [llvm/llvm-project#67213](https://github.com/llvm/llvm-project/pull/67213)
> 
> That still doesn't show the total padding of the struct? :(

No, because struct size shows with paddings. Also, you can use `-WPadded` flag for it

https://github.com/clangd/clangd/issues/923#issuecomment-1263681682

**Comment by medhefgo** [comment]
> No, because struct size shows with paddings. Also, you can use `-WPadded` flag for it
> 
> [#923 (comment)](https://github.com/clangd/clangd/issues/923#issuecomment-1263681682)

Yes, I know that padding is part of the struct size. But there is value in knowing if/how much padding it has when I create a struct. And no, `-Wpadded` is not an option. I don't want a warning for every damn struct I see that has some padding.
